### ISSUE TITLE
[Usage]: slow inference for fine-tuned model

### ISSUE THREAD

**Issue by yananchen1989** [vllm-project/vllm#4243]
### Your current environment

I am using SFT of huggingface transformers to fine tune model mistral 7b. https://github.com/huggingface/trl/blob/main/examples/scripts/sft.py

After fine-tuning, the model works well, by loading via langchain - vllm  (vllm version: 0.4.0)(https://python.langchain.com/docs/integrations/llms/vllm/), 
```
from langchain_community.llms import VLLM
llm = VLLM(
    model="mistralai/Mistral-7B-Instruct-v0.2", # or drop-in replacement of fine-tuned model with local path
    trust_remote_code=True,  # mandatory for hf models
    max_new_tokens=128,
    top_k=10,
    top_p=0.95,
    temperature=0.8,
)
```
However, the inference latency is round 2-3 times slower than the original "mistralai/Mistral-7B-Instruct-v0.2".
May I ask is there any specific settings that i have missed for applying vllm on SFT model ?
Thanks.

### How would you like to use vllm

run vllm on fine-tuned model.

**Comment by yananchen1989** [comment]
this issue also exist in the original vllm inference api, not only exist for langchain api.

**Comment by skandermoalla** [comment]
@youkaichao @qgallouedec  I'm running into the same issue with 2 different models. I tried with `mistralai/Mistral-7B-Instruct-v0.2`as mentioned above and with `meta-llama/Meta-Llama-3-8B`

The speed input is 4x lower.

With the checkpoint before sft I have these stats
```
Processed prompts:   0%|          | 100/26240 [22:43<107:27:47, 14.80s/it, est. speed input: 3.25 toks/s, output: 7128.51 toks/s]
```

with the checkpoint after the sft I have these stats
```
Processed prompts:   0%|          | 116/26240 [1:43:25<373:53:00, 51.52s/it, est. speed input: 0.89 toks/s, output: 6476.62 toks/s
```

I have checked that nothing is coming from the saved configs, and by replacing only the safetensors in the sft checkpoint by the non-sft checkpoint, I could recover the performance.

I don't know how to proceed to investigate further. I can send the checkpoints.

I trained the sft model with the `SFTTrainer` with accelerate with the deepspeed plugin at stage 1.

Any help is welcome.
